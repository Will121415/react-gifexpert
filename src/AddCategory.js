import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ( {setCategory} ) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (event) => {
        setInputValue(event.target.value);
    }

    const handleSubmit = (event) => {
        event.preventDefault();
        if(inputValue.trim().length > 2) {
            setCategory(category => [inputValue, ...category]);
            setInputValue('');
        }
    }

    return (
        <form onSubmit={handleSubmit}>
            <input
                type="text"
                value={inputValue}
                onChange={handleInputChange}
            />
        </form>
    )

}

AddCategory.propTypes = {
    setCategory: PropTypes.func.isRequired
}
