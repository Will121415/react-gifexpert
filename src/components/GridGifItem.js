import React from 'react'
import { PropTypes } from 'prop-types'
export const GridGifItem = ({url,title}) => {
  return (
    <div className='card'>
        <img src={url} alt={title} />
        <p>{title}</p>
    </div>
  )
}

GridGifItem.propTypes = {
  url: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired
}