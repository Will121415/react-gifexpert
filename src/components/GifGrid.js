import React from 'react'
import { GridGifItem } from './GridGifItem';
import { useFetchGifs } from '../Hooks/useFetchGifs';
import { PropTypes } from 'prop-types';

export const GifGrid = ({ category }) => {

    const {data:images, loading} = useFetchGifs(category);

    // console.log(category, images, loading);


    return (
        <>
            <h3>{category}</h3>
            {loading && <strong>Loading...</strong>}
            <div className='card-grid animate__animated animate__fadeIn'>
                {
                    images.map(img => (
                        <GridGifItem
                            key={img.id}
                            {...img}
                        />
                    ))
                }
            </div>
        </>

    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}