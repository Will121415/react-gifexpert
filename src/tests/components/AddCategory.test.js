import React from 'react';
import { shallow } from 'enzyme'; 
import { AddCategory } from '../../components/AddCategory';

describe('Pruebas en el componente <AddCategory />', () => {

    const setCategories = jest.fn();
    let wapper = shallow(<AddCategory setCategory={ setCategories } />);

    beforeEach(() => {
        jest.clearAllMocks();
        wapper = shallow(<AddCategory setCategory={ setCategories } />);
    })

    test('Debe funcionar correctamente', () => { 

        expect(wapper).toMatchSnapshot();
    })

    test('debe de cambiar la caja de texto', () => {
            
            const input = wapper.find('input');
            const value = 'Hola Mundo';
    
            input.simulate('change', { target: { value } });
    
            expect(wapper.find('p').text().trim()).toBe(value);
    })

    test('No debe postear la info con Submit si inputValue esta vacio', () => {

        wapper.find('form').simulate('submit',{ preventDefault() {} });
        expect(setCategories).not.toHaveBeenCalled();

    })

    test('Debe llamar el setCategory y limpiar el inputValue', () => {

        const value = 'Hola Mundo';
        wapper.find('input').simulate('change', { target: {value}});
        wapper.find('form').simulate('submit', { preventDefault() {} });

        expect(setCategories).toHaveBeenCalled();
        expect( setCategories ).toHaveBeenCalledWith( expect.any(Function) );

        expect(wapper.find('input').prop('value')).toBe('');
    })

})