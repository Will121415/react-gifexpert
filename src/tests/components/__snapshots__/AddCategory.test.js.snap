// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <AddCategory /> Debe funcionar correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
